File: code/backend/TA-API/Models/TaskItem.cs
namespace TA_API.Models
{
    /// <summary>
    /// Domain entity representing a task item.
    /// </summary>
    public class TaskItem
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public required string Description { get; set; }
        public bool IsCompleted { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public required string UserId { get; set; }
    }
}

File: code/frontend/react/src/types/TaskItem.js
/**
 * TaskItem domain types matching the backend API
 * @typedef {Object} TaskItem
 * @property {number} id
 * @property {string} title
 * @property {string} description
 * @property {boolean} isCompleted
 * @property {string} createdAt
 * @property {string} userId
 */

/**
 * @typedef {Object} CreateTaskItemRequest
 * @property {string} title
 * @property {string} description
 * @property {string} userId
 */

/**
 * @typedef {Object} UpdateTaskItemRequest
 * @property {string} title
 * @property {string} description
 * @property {boolean} isCompleted
 */

/**
 * @typedef {Object} ApiResponse
 * @property {any} data
 * @property {string} [error]
 * @property {number} statusCode
 */

// Exports are handled naturally with JSDoc comments
